pkg/resultstore/store.go

```diff
@@ -0,0 +1,189 @@
+package resultstore
+
+import (
+	"context"
+	"log"
+	"sort"
+	"sync"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+// TODO: make these configurable?
+var (
+	notifyQBufferSize = 128
+	storeTTL          = time.Minute
+	gcInterval        = time.Minute * 5
+)
+
+// element is an internal representation of a result.
+type element struct {
+	data    ocr2keepers.CheckResult
+	addedAt time.Time
+}
+
+// resultStore implements ResultStore.
+type resultStore struct {
+	lggr *log.Logger
+
+	data map[string]element
+	lock sync.RWMutex
+
+	notifications chan ocr2keepers.Notification
+}
+
+func New(lggr *log.Logger) *resultStore {
+	return &resultStore{
+		lggr:          lggr,
+		data:          make(map[string]element),
+		lock:          sync.RWMutex{},
+		notifications: make(chan ocr2keepers.Notification, notifyQBufferSize),
+	}
+}
+
+// Start starts the store, it spins up a goroutine that runs the garbage collector every gcInterval.
+func (s *resultStore) Start(pctx context.Context) error {
+	s.lggr.Println("starting result store")
+
+	go func() {
+		ctx, cancel := context.WithCancel(pctx)
+		defer cancel()
+
+		ticker := time.NewTicker(gcInterval)
+		for {
+			select {
+			case <-ticker.C:
+				s.gc()
+			case <-ctx.Done():
+				s.lggr.Println("result store context done, stopping gc")
```

Comment:
Should you call `ticker.Stop()` here? Or defer a call to it on line 54?  Otherwise we might have a memory leak?